tests/spki_tests.rs

```diff
@@ -0,0 +1,615 @@
+extern crate ring;
+extern crate untrusted;
+extern crate rustc_serialize;
+
+use std::io::BufRead;
+use std::path::Path;
+use std::vec::Vec;
+use std::io::Read;
+use std::fs::File;
+use rustc_serialize::base64::FromBase64;
+use ring::der;
+use ring::signature::spki;
+use ring::signature::spki::VerifyWithSPKIError;
+use ring::signature;
+#[cfg(feature = "rsa_signing")]
+use ring::rand;
+
+// TODO: The expected results need to be modified for SHA-1 deprecation.
+
+macro_rules! test_verify_signature_pem {
+    ($fn_name:ident, $file_name:expr, $signature_alg:expr, $expected_result:expr) => {
+        #[test]
+        fn $fn_name() {
+            test_verify_signature_pem($file_name, $signature_alg, $expected_result);
+        }
+    }
+}
+
+fn test_verify_signature_pem(file_name: &str,
+                             signature_algorithm: &'static spki::Algorithm,
+                             expected_result: Result<(), VerifyWithSPKIError>) {
+    let tsd = parse_test_signed_data(file_name);
+    let spki_value = untrusted::Input::from(&tsd.spki);
+    let spki_value = spki_value.read_all(VerifyWithSPKIError::BadDER, |input| {
+        der::expect_tag_and_get_value(input, der::Tag::Sequence)
+            .map_err(|_| VerifyWithSPKIError::BadDER)
+    }).unwrap();
+
+    let signature = untrusted::Input::from(&tsd.signature);
+    let signature = signature.read_all(VerifyWithSPKIError::BadDER, |input| {
+        der::bit_string_with_no_unused_bits(input)
+            .map_err(|_| VerifyWithSPKIError::BadDER)
+    }).unwrap();
+
+    assert_eq!(expected_result,
+    spki::verify(signature_algorithm,
+                 spki_value,
+                 untrusted::Input::from(&tsd.data),
+                 signature));
+}
+
+macro_rules! test_rsa_sign_verify_spki {
```

Comment:
No ECDSA signing support yet, so we don't have an equivalent ECDSA "sign then verify the signature" scenario.